code/iaas/agent-instance/src/main/java/io/cattle/platform/agent/instance/dao/impl/AgentInstanceDaoImpl.java

```diff
@@ -63,19 +75,51 @@ public Instance getInstanceByAgent(Agent agent) {
 
     @Override
     public List<Long> getAgentProvider(String providedServiceLabel, long accountId) {
-        return Arrays.asList(create().select(INSTANCE.AGENT_ID)
+        String priorityLabel = providedServiceLabel + ".priority";
+        final LabelTable priority = LABEL.as("priority");
+        final Map<Long, Integer> result = new HashMap<>();
+        create().select(INSTANCE.AGENT_ID, priority.VALUE)
                 .from(INSTANCE)
                 .join(INSTANCE_LABEL_MAP)
                     .on(INSTANCE_LABEL_MAP.INSTANCE_ID.eq(INSTANCE.ID))
                 .join(LABEL)
                     .on(LABEL.ID.eq(INSTANCE_LABEL_MAP.LABEL_ID).and(LABEL.KEY.eq(providedServiceLabel)))
+                .leftOuterJoin(priority)
+                    .on(LABEL.ID.eq(INSTANCE_LABEL_MAP.LABEL_ID).and(LABEL.KEY.eq(priorityLabel)))
                 .where(INSTANCE.ACCOUNT_ID.eq(accountId)
                     .and(INSTANCE.AGENT_ID.isNotNull())
                         .and(INSTANCE.STATE.eq(InstanceConstants.STATE_RUNNING))
                         .and(INSTANCE.HEALTH_STATE.in(HealthcheckConstants.HEALTH_STATE_HEALTHY,
```

Comment:
@cjellick This imposes the requirement of a health check on any external scheduler. Is that OK?

I ask because the windows scheduler is running in host networking mode without a health check.